`TestExecKubeService ` flakiness

## Failure

#### Link(s) to logs

https://github.com/gravitational/teleport.e/actions/runs/15873361431/job/44755034387

#### Relevant snippet


```
    exec_test.go:246: 
        	Error Trace:	/__w/teleport.e/teleport.e/lib/kube/proxy/exec_test.go:246
        	Error:      	Not equal: 
        	            	expected: "teleportContainer\nstdin_data"
        	            	actual  : "stdin_data"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1 @@
        	            	-teleportContainer
        	            	 stdin_data
        	Test:       	TestExecKubeService/SPDY_protocol
--- FAIL: TestExecKubeService/SPDY_protocol (0.09s)
```

--- a/lib/kube/proxy/testing/kube_server/kube_mock.go
+++ b/lib/kube/proxy/testing/kube_server/kube_mock.go
@@ -393,14 +393,23 @@ func (s *KubeMockServer) exec(w http.ResponseWriter, req *http.Request, p httpro
 	}
 	defer proxy.Close()
 
+	var outStream, errStream io.Writer
+	if request.tty {
+		outStream = proxy.stdoutStream
+		errStream = proxy.stderrStream
+	} else {
+		outStream = bytes.NewBuffer(nil)
+		errStream = bytes.NewBuffer(nil)
+	}
+
 	if request.stdout {
-		if _, err := proxy.stdoutStream.Write([]byte(request.containerName + "\n")); err != nil {
+		if _, err := outStream.Write([]byte(request.containerName + "\n")); err != nil {
 			s.log.ErrorContext(request.context, "unable to send to stdout", "error", err)
 		}
 	}
 
 	if request.stderr {
-		if _, err := proxy.stderrStream.Write([]byte(request.containerName + "\n")); err != nil {
+		if _, err := errStream.Write([]byte(request.containerName + "\n")); err != nil {
 			s.log.ErrorContext(request.context, "unable to send to stderr", "error", err)
 		}
 	}
@@ -430,19 +439,26 @@ func (s *KubeMockServer) exec(w http.ResponseWriter, req *http.Request, p httpro
 			}
 
 			if request.stdout {
-				if _, err := proxy.stdoutStream.Write(buffer); err != nil {
+				if _, err := outStream.Write(buffer); err != nil {
 					s.log.ErrorContext(request.context, "unable to send to stdout", "error", err)
 				}
 			}
 
 			if request.stderr {
-				if _, err := proxy.stderrStream.Write(buffer); err != nil {
-					s.log.ErrorContext(request.context, "unable to send to stdout", "error", err)
+				if _, err := errStream.Write(buffer); err != nil {
+					s.log.ErrorContext(request.context, "unable to send to stderr", "error", err)
 				}
 			}
-
 		}
+	}
 
+	if !request.tty {
+		if _, err := io.Copy(proxy.stdoutStream, outStream.(*bytes.Buffer)); err != nil {
+			s.log.ErrorContext(request.context, "unable to copy to stdout", "error", err)
+		}
+		if _, err := io.Copy(proxy.stderrStream, errStream.(*bytes.Buffer)); err != nil {
+			s.log.ErrorContext(request.context, "unable to copy to stderr", "error", err)
+		}
 	}
 
 	return nil, nil

